packages/flame_svg/lib/svg.dart

```diff
@@ -80,9 +80,12 @@ class Svg {
   /// this method once the instance is no longer needed to avoid
   /// memory leaks
   void dispose() {
+    final disposed = <Size>[];
```

Comment:
Hmm, we'd preferably wouldn't want to create a new object in here.
Can't the call be made directly inside `forEach`?

Reply:
We can't mutate the list while iterating it using `forEach`. Changing it to a `while` loop could help.

Reply:
Oh, we can use `clear` function to clear all items after the loop. Don't need to remove each item one by one.

Reply:
The naming is slightly confusing for it, so `clearCache` should be used then, but it would probably be could to add `clear(key)` too since that exists in the other cache.

Reply:
Pardon me, I meant the `clearCache` function 😄 Hmm, maybe I will add them all and write tests for them as well.

Reply:
Added both of them and more tests.